students/ghassan/lesson04/new_mailroom.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+
+import sys
+
+global donors
+donors = {
+        'Jessie': [100, 300, 500],
+        'James': [343, 22.11],
+        'Ford': [3000, 400]}
+
+
+def main_menu():
+    choice = input(
+        'Pick one:\n1) Send a thank you\n2) Create a Report\n\
+3) Send letters to everyone\n4) Quit\n')
+    return str(choice)
+
+
+def send_thankyou():
+    donor_name = input('Enter the donor\'s name: ')
+    if donor_name == 'list':
+        print(donor.keys())
+    elif donor_name not in donors.keys():
+        donors[donor_name] = []
+    donation_amount = input(
+        'Please enter a donation amount: ')
+    donors[donor_name].append(donation_amount)
+    print('Thank you {} for yout generous donation of {}'.format(
+        donor_name, donation_amount))
+
+
+def create_report():
+    print('{:20} | {:15} | {:10} | {:15}'.format(
+        'Donor Name', 'Total Given', 'Num Gifts', 'Average Gift'))
+    print('-'*70)
+    for donor, amounts in donors.items():
+        print('{:20} | {:15} | {:10} | {:15}'.format(
+            donor, sum(amounts),
+            len(amounts),
+            sum(amounts) / len(amounts)))
+
+
+def save_report():
+    for donor, amounts in donors.items():
+        with open(donor+'.txt', 'w') as donorfh:
+            donorfh.write(
+                'Thank you {} for yout generous donation of {}\n'.format(
+                    donor, sum(amounts)))
+
+
+def quittt():
+    sys.exit()
+
+
+def main():
+    users_choice = main_menu()
```

Comment:
The program needs to return to the main prompt and continue running after exiting each selection. A `while` loop is a great construct to use for this purpose.